Dockerfile

```diff
@@ -1,20 +1,197 @@
+ && cd ../dpdk1911_x86_64 \
+ && cp /tmp/dpdk/build/include/rte_config.h . \
+ && patch < /tmp/rte_config.h.patch \
+    # Apply the DPDK Trex patch that we built earlier.
+ && cd /tmp/trex-core/src/dpdk/lib \
+ && patch -p1 < /tmp/libs.patch || true \
+ && cd librte_mbuf \
+ && patch rte_mbuf_core.h < rte_mbuf.h.rej \
+ && cd ../../drivers/net/mlx5 \
+ && patch < /tmp/mlx5_flow.c.patch \
+ && cd ../../../../ \
+    # A bunch of crap was renamed and moved. Deal with that using sed.
+ && find . -path ./dpdk -prune -o -type f -print | xargs sed -i '/DEV_TX_OFFLOAD_MATCH_METADATA/d' \
+ && find . -path ./dpdk -prune -o -type f -print | xargs sed -i 's/ETHER_ADDR_FMT_SIZE/RTE_ETHER_ADDR_FMT_SIZE/g' \
+ && find . -path ./dpdk -prune -o -type f -print | xargs sed -i 's/ETHER_TYPE_ARP/RTE_ETHER_TYPE_ARP/g' \
+ && find . -path ./dpdk -prune -o -type f -print | xargs sed -i 's/ETHER_TYPE_IPV4/RTE_ETHER_TYPE_IPV4/g' \
+ && find . -path ./dpdk -prune -o -type f -print | xargs sed -i 's/ETHER_TYPE_IPV6/RTE_ETHER_TYPE_IPV6/g' \
+ && find . -path ./dpdk -prune -o -type f -print | xargs sed -i 's/ETHER_TYPE_IPv4/RTE_ETHER_TYPE_IPV4/g' \
+ && find . -path ./dpdk -prune -o -type f -print | xargs sed -i 's/ETHER_TYPE_IPv6/RTE_ETHER_TYPE_IPV6/g' \
+ && find . -path ./dpdk -prune -o -type f -print | xargs sed -i 's/ETHER_TYPE_VLAN/RTE_ETHER_TYPE_VLAN/g' \
+ && find . -path ./dpdk -prune -o -type f -print | xargs sed -i 's/ETHER_TYPE_QINQ/RTE_ETHER_TYPE_QINQ/g' \
+ && find . -path ./dpdk -prune -o -type f -print | xargs sed -i 's/ ipv4_hdr / rte_ipv4_hdr /g' \
+ && find . -path ./dpdk -prune -o -type f -print | xargs sed -i 's/ ipv6_hdr / rte_ipv6_hdr /g' \
+ && find . -path ./dpdk -prune -o -type f -print | xargs sed -i 's/ tcp_hdr / rte_tcp_hdr /g' \
+ && find . -path ./dpdk -prune -o -type f -print | xargs sed -i 's/ udp_hdr / rte_udp_hdr /g' \
+ && find . -path ./dpdk -prune -o -type f -print | xargs sed -i 's/ ether_addr / rte_ether_addr /g' \
+ && find . -path ./dpdk -prune -o -type f -print | xargs sed -i 's/eal_parse_pci_BDF/rte_pci_addr_parse/g' \
+ && find . -path ./dpdk -prune -o -type f -print | xargs sed -i 's/eal_parse_pci_DomBDF/rte_pci_addr_parse/g' \
+ && find . -path ./dpdk -prune -o -type f -print | xargs sed -i 's/rte_eth_dev_count(/rte_eth_dev_count_avail(/g' \
+ && find . -path ./dpdk -prune -o -type f -print | xargs sed -i 's/ether_format_addr/rte_ether_format_addr/g' \
+    # This static assertion triggers. The mbuf size must have changed.
+ && sed -i '/size of MBUF must be 128/d' pal/linux_dpdk/mbuf.cpp \
+ && cd ../linux_dpdk \
+    # Actually build and install Trex.
+ && ./b configure \
+ && ./b build \
+ && ./b install \
+ && ldconfig
 
-COPY traffic-configurations/veth_cfg.yaml /etc/trex_cfg.yaml
-COPY traffic-configurations/*.yaml /tmp/trex/cfg/
-COPY traffic-patterns/dns_customer_ip.yaml /tmp/trex/cap2/dns_customer_ip.yaml
+# Copy our Trex configuration files into the image.
+COPY traffic-configurations/mlx_cfg.yaml /etc/trex_cfg.yaml
```

Comment:
I understand making mlx_cfg.yaml the default config file for node100, but can we still copy the other yaml files to a working directory so that the docker container can be utilized still on a local machine? Or did you want to remove support for making things work on a local container?

Reply:
This isn't removing support. It's just defaulting it to use the mlx5 hardware. We don't need to copy the files in, you just need to use `--volume somthing.yaml:/etc/trex_cfg.yaml` to use a different config. We can add the ability to swap out configs in the launch script, but we don't need to do that immediately. I would also prefer to just make people volume mount in their `traffic-patterns`. Neither of these should be embedded in the image, but I think what's here is "good enough."

Reply:
And since the `traffic-patterns` directory is still around, there's at least a simple way to mount that in. Ok, I'm happy to step back here.